src/Widgets/Banner.vala

```diff
@@ -102,24 +102,28 @@ public class AppCenter.Widgets.Banner : Gtk.Button {
         add (package_grid);
 
         var provider = new Gtk.CssProvider ();
-        var background_color = package.get_color_primary ();
-        var foreground_color = package.get_color_primary_text ();
+        try {
+            string? bg_color = null;
+            string text_color = DEFAULT_BANNER_COLOR_PRIMARY_TEXT;
 
-        if (background_color == null || foreground_color == null) {
-            background_color = DEFAULT_BANNER_COLOR_PRIMARY;
-            foreground_color = DEFAULT_BANNER_COLOR_PRIMARY_TEXT;
-        }
+            if (package != null) {
+                bg_color = package.get_color_primary ();
 
-        var colored_css = BANNER_STYLE_CSS.printf (
-            background_color,
-            foreground_color
-        );
+                if (bg_color == null) {
+                    bg_color = DEFAULT_BANNER_COLOR_PRIMARY;
+                } else {
+                    var bg_rgba = Gdk.RGBA ();
+                    bg_rgba.parse (bg_color);
 
-        try {
+                    text_color = Granite.contrasting_foreground_color (bg_rgba).to_string ();
+                }
+            }
+
+            var colored_css = BANNER_STYLE_CSS.printf (bg_color, text_color);
```

Comment:
Is it OK to (potentially) print `null` into the css here?

Reply:
Oops you're right, that is problematic. Should be fixed now